File: Spang Java Core/src/serialization/StringSerializer.java
package serialization;

import serialization.Serializer;
import utils.Packer;
import utils.UnPacker;

public class StringSerializer extends Serializer<String>{

	public Class<?> getSerializableType() {
		return String.class;
	}

	@Override
	protected void serializeInternal(Packer packer, String message) {
		packer.packString(message);
	}

	@Override
	public String deserialize(UnPacker unpacker) {
		return unpacker.unpackString();
	}

}

File: Spang-PC_C-sharp/Spang.Core/Serialization/StringSerializer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Spang.Core.Utils;

namespace Spang.Core.Serialization
{
    public class StringSerializer : Serializer<String> 
    {
        public override void SerializeInternal(Packer packer, string message)
        {
            packer.Pack(message);
        }

        public override string DeserializeInternal(UnPacker unpacker)
        {
            return unpacker.UnpackString();
        }
    }
}
